codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -277,6 +278,23 @@ public boolean visit(Http2Stream stream) {
         }
     }
 
+    @Override
+    public void headersSent(int streamId, boolean sent) {
+        Http2Stream stream = stream(streamId);
+        assert stream != null : "Invalid stream id";
+        stream.setProperty(headersSentKey, sent);
+    }
+
+    @Override
+    public boolean headersSent(int streamId) {
+        Http2Stream stream = stream(streamId);
+        if (stream == null) {
+            return false;
```

Comment:
I see how this makes sense from a generic interface, but in this particular case, if we can't get a stream object, we are not going to send any headers anyway.